### DISCUSSION TITLE
About Airflow Drag and Drop Extention

### DISCUSSION THREAD

**Discussion by emincanoguz11** [apache/airflow#26516] [Ideas]
Hi there,

My name is Emin Can. I am a Data Engineer at Hepsiburada in Turkey. I have an idea for Airflow as a beginner.  First of all I loved write dag but dag is more complex than drag and drop extension. So that I want to contribute this but first I want to take your thoughts. This extension similar like Oozie Workflow. Oozie is very simple for non-technique person and beginners. Is this good for Airflow or is it not necessary for beginners.

I am waiting your thoughts.
Best wishes,
Emin Can

**Comment by potiuk** [comment]
This is not new idea/discussion.

There are lots of discussions about it and it's something tha many extensions already exist about https://cwl-airflow.readthedocs.io/en/latest/ , some other solution - for example video from here https://airflowsummit.org/sessions/demo-visual-dag-editor/  and a dosen of others.

We are not currently considering this as important to be part of the open-source airflow (we build the "foundation" for dag execution and our DAGs are python code by default, definition and conscious choice. 

Howerver we know at some point of time some simpler way of authoring DAGs might be helpful to become part of Airflow and it is likely to happen, but this will have to be preceeded by a long discussion on the devllst and writing, discussing and getting community consent on Airflow Improvement Proposal: https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals. 

It's rather impossible you can just "contribute" it just like that. This will be a very long debated what/how/if/when/why and likely big number of people will have to be involved and they will have their own opinions and ideas, so this is rather a marathon not sprint.

I am not discouraging you from contributing it, just making you aware that there will likely be many months between the willingness to start on it is raised on the devlist and the first line of code merged.

**Reply by emincanoguz11** [reply]
Hi @potiuk 

First of all thanks for detail answer. I want to discuss extension for this project but I think you are right. I understand that. I think about this perspective for improving users experience. This project is big and lots of users are using Airflow. Honestly this tool is very cool and I love Airflow. Has Airflow only cwiki page for discussion session or has Airflow community channel such as Slack, Discord or e mail groups? Maybe we discuss this extension and writing the Improvement Proposal. 

Again thanks for answer.

**Reply by potiuk** [reply]
Sure: you will find all the information about community here: https://airflow.apache.org/community/
